### PR TITLE
Adding OAuth Support

### PR DESCRIPTION
**Pull request by VasudhaMathur**

### CODE REVIEW

File: src/authentication.cpp [hunk 1]
```diff
@@ -1,29 +1,51 @@
...
     authKeys["credentialToken"] = m_client_id;
     authKeys["credentialSecret"] = m_client_secret;
 
-    Ruqola::self()->ddp()->method("login", QJsonDocument(authKeys));
+    QJsonArray requestPermissions;
+    requestPermissions.append("email");
+
+    bool requestOfflineToken = true;
+
+    QString scope = QString("%1 %2 %3").arg("openID").arg("profile").arg("email");
```
Comment (ruphy) [inline src/authentication.cpp]: do you really need .arg? can't you just: QString scope("openID profile email") ?

File: src/authentication.cpp [hunk 2]
```diff
@@ -1,29 +1,51 @@
...
+
+    QSettings s;
+    s.setValue("stateHexNumber", QString("{67C8770B-44F1-410A-AB9A-F9B5446F13EE}"));
```
Comment (ruphy) [inline src/authentication.cpp]: what's this value?

File: src/client_secret.json [hunk 3]
```diff
@@ -1 +1 @@
-{"web":{"client_id":"143580046552-s4rmnq5mg008u76id0d3rl63od985hc6.apps.googleusercontent.com","project_id":"ruqola-161705","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://accounts.google.com/o/oauth2/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"nyVm19iOjjtldcCZJ-7003xg","redirect_uris":["http://localhost:8080/cb","http://127.0.0.1:3000"]}}
\ No newline at end of file
+{"web":{"client_id":"143580046552-s4rmnq5mg008u76id0d3rl63od985hc6.apps.googleusercontent.com","project_id":"ruqola-161705","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://accounts.google.com/o/oauth2/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"nyVm19iOjjtldcCZJ-7003xg","redirect_uris":["http://localhost:8080/cb/_oauth/google?close","http://localhost:8080/cb"]}}
```
Comment (ruphy) [inline src/client_secret.json]: OK so this is a development key? not sure how to keep this information private in an open source world

Comment (VasudhaMathur) [inline src/client_secret.json]: We can add them to gitignore (better than nothing).

File: src/authentication.cpp (hunk 2 above)
Comment (VasudhaMathur) [inline src/authentication.cpp]: https://developers.google.com/identity/protocols/OpenIDConnect#authenticationuriparameters
The state parameter. We can take any random hex value so this is just any random value for now :p

File: src/client_secret.json (hunk 3 above)
Comment (ruphy) [inline src/client_secret.json]: This one you just added is compromised though, it went live. Let's keep using this for development, we will fix this later

File: src/authentication.cpp (hunk 2 above)
Comment (ruphy) [inline src/authentication.cpp]: OK, we should still randomize it though, according to the documentation :) so add a todo (or just do it straight away)

File: src/client_secret.json (hunk 3 above)
Comment (VasudhaMathur) [inline src/client_secret.json]: Yes of course. We can reset it anytime we wish :)

**Pull request merged**